src/app/services/xml-parsers/apparatus-entries-parser.service.ts

```diff
@@ -0,0 +1,128 @@
+    map((source) => this.parseAppEntriesList(source)),
+    shareReplay(1),
+  );
+
+  private appEntryTagName = 'app';
+  private lemmaTagName = 'lem';
+  private readingTagName = 'rdg';
+  private readingGroupTagName = 'rdgGrp';
+  private noteTagName = 'note';
+  private significantRdgNumber = 0;
+  private parsedWitnesses = {};
+
+  constructor(
+    private editionDataService: EditionDataService,
+    private genericParserService: GenericParserService,
+    private witParserService: WitnessesParserService,
+  ) {
+  }
+
+  private parseAppEntriesList(document: XMLElement): ApparatusEntriesList {
+    const appEntries = Array.from(document.querySelectorAll<XMLElement>(this.appEntryTagName));
+
+    return this.parseAppEntries(appEntries);
+  }
+
+  private parseAppEntries(appEntries: XMLElement[]) {
+    this.witParserService.witnesses$.subscribe(data => this.parsedWitnesses = data);
+    const parsedEntries = appEntries.map((appEntry) => this.parseAppEntry(appEntry));
+
+    return arrayToMap(parsedEntries, 'id');
+  }
+
+  private parseAppEntry(appEntry: XMLElement): ApparatusEntry {
+    this.significantRdgNumber = 0;
+
+    return {
+      type: 'ApparatusEntryComponent',
+      id: appEntry.getAttribute('xml:id') || xpath(appEntry),
+      attributes: this.genericParserService.parseAttributes(appEntry),
+      content: this.parseAppReadings(appEntry),
+      note: this.genericParserService.parse(appEntry.querySelector<XMLElement>(this.noteTagName)),
+      variance: this.getVariance(),
+      originalEncoding: getOuterHTML(appEntry),
+    };
+  }
+
+  private parseAppReadings(appEntry: XMLElement): ApparatusReading[] {
+    return Array.from(appEntry.querySelectorAll(`${this.lemmaTagName}, ${this.readingTagName}`))
+      .map((child: XMLElement) => this.parseAppReading(child));
+  }
+
+  private parseAppReading(rdg: XMLElement): ApparatusReading {
+    return {
+      type: 'ApparatusReadingComponent',
+      tagname: rdg.tagName,
+      id: rdg.getAttribute('xml:id') || xpath(rdg),
+      attributes: this.genericParserService.parseAttributes(rdg),
+      wits: this.parseReadingWitnesses(rdg) || [],
+      content: this.parseAppReadingContent(rdg),
+      significant: this.parseReadingSignificance(rdg),
+    };
+  }
+
+  private parseReadingWitnesses(rdg: XMLElement) {
+    return rdg.getAttribute('wit')?.split('#')
+      .map((el) => removeSpaces(el))
+      .filter((el) => el.length !== 0);
+  }
+
+  private parseAppReadingContent(rdg: XMLElement) {
+    return Array.from(rdg.childNodes)
+      .map((child: XMLElement) => {
+        if (child.nodeName === this.appEntryTagName) {
+          return {
+            type: 'ApparatusEntryComponent',
+            id: child.getAttribute('xml:id') || xpath(child),
+            attributes: {},
+            content: [],
+          };
+        }
+
+        return this.genericParserService.parse(child);
+      });
+  }
+
+  private parseReadingSignificance(rdg: XMLElement) {
```

Comment:
I would organize differently the two functions `parseReadingSignificance ` and `isSignificantReading `. The second one should just check if it is significant and return the boolean. I would probably put the other computations inside `parseReadingSignificance` since its name suggests that besides checking if is significant you also do some other stuff with that info.

Reply:
Done

Reply:
What's the meaning of "significant" reading?

Reply:
"Significant reading" is in contrast with "non-significant reading", that is, a reading that does not present a particular variance with respect to the lemma. For example an ortographic variation is not as important as a lexicographic variation. We recover the "non-significant reading" from the config file.

Reply:
In traditional printed critical edition non-significant reading where usually excluded in order to save space. In digital edition they can now be encoded and visualized since there are no space problems anymore; but in a situation with a very high number of witnesses and a very high variance is sometimes preferable to hide them. That's why we've decided to distinguish them.